Automattic/wp-calypso
# Simplify the KeyBindings module and its translations

This patch simplifies the `KeyBindings` module and converts it into a simple export of a constant static data structure.

The `KeyBindings` object is used to initialize the global keyboard shortcuts (that's why it's loaded on Calypso boot) and also to populate the keyboard shortcuts modal that appears after pressing 'Shift-/'.

The description strings are no longer statically translated when evaluating the module, but instead we're exporting functions in the form:
```js
translate => translate( 'Shortcut Description' )
```
as the `description.text` fields. This allows to translate the strings using the standard `localize` HOC. The translations are no longer done on Calypso boot, but instead only when the modal is displayed. That removes a few milliseconds from boot time, but also simplifies the module: for example, we no longer need to listen for the `change` event from the `i18n` object and fire `language-change`
events from `KeyBindings`. Also, the return value from `KeyBindings.get` is not re-created on each call, but it's a static object directly exported.

I also removed the test for `key-bindings.js`, because it became tautological -- just testing a constant static export.

**How to test:**
- check that all keyboard shortcuts still work as before
- check that the modal with shortcuts description appears after 'Shift-/'. Beware that this feature is turned off on staging and production, and only enabled in development and wpcalypso/horizon.
